lfsapi/ntlm.go

```diff
@@ -163,4 +125,21 @@ func rewoundRequestBody(req *http.Request) (io.ReadCloser, error) {
 	return body, err
 }
 
-const ntlmNegotiateMessage = "NTLM TlRMTVNTUAABAAAAB7IIogwADAAzAAAACwALACgAAAAKAAAoAAAAD1dJTExISS1NQUlOTk9SVEhBTUVSSUNB"
+func ntlmGetCredentials(creds Creds) (ntmlCredentials, error) {
```

Comment:
I think going along with my suggestion about this method could return a `(*ntlmCredentials, error)`, which would contain a 👍 return value indicating the absence of NTLM credentials.

Reply:
@ttaylorr thanks so much for reviewing the code. I really appreciate your feedback.

I don't quite understand what you mean by "which would contain a (thumbsup) return value". Do you mean that we could return `(nil, nil)` instead of `(&ntlmCredentials{}, nil) in case no NTLM credentials were provided? This would also mean we could get rid of `IsDefaultCredentials`, so no NTLM credentials would for windows simply mean use the default ones of the current user.

Reply:
> I don't quite understand what you mean by "which would contain a (thumbsup) return value". Do you mean that we could return `(nil, nil)` instead of `(&ntlmCredentials{}, nil)` in case no NTLM credentials were provided?

Your intuition is totally correct -- and my apologies for not being as clear as I could have been. My intention was to suggest that `nil` is a clearer representation of "I couldn't find credentials" than an empty `ntlmCredentials` instance.

> This would also mean we could get rid of `IsDefaultCredentials`, so no NTLM credentials would for windows simply mean use the default ones of the current user.

Indeed 👍.